src/main/java/com/aws/greengrass/lifecyclemanager/PluginService.java

```diff
@@ -14,10 +14,12 @@
 import java.net.URLClassLoader;
 import java.util.Map;
 import javax.inject.Inject;
+import javax.inject.Singleton;
 
 import static com.aws.greengrass.componentmanager.KernelConfigResolver.VERSION_CONFIG_KEY;
 import static com.aws.greengrass.deployment.bootstrap.BootstrapSuccessCode.REQUEST_RESTART;
 
+@Singleton
```

Comment:
does `PluginService` itself ever get instantiated? Or do classes just extend it?

Reply:
Only extending classes.

Reply:
But the annotation should apply to extending classes, right?

Reply:
The singleton annotation is not inherited by subclasses - annotations can have an `@Inherited` meta-annotation on them - but singleton does not

Reply:
OK thanks, I reverted it. Will look into it later.